### DISCUSSION TITLE
Welcome to WeBAD Discussions!

### DISCUSSION THREAD

**Discussion by solyarisoftware** [solyarisoftware/WeBAD#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by hansalbertisaksson** [comment]
Hello everyone,

Firstly, I'd like to extend my gratitude for open-sourcing WeBAD!

As a hobbyist developer, I've stumbled upon an interesting use case that aligns with my current project. I'm attempting to integrate WeBAD with the OpenAI Whisper API to transcribe text from audio. Given that Whisper does not natively support streamed audio transcription, I found that WeBAD could fill this gap perfectly.

Here's my current approach: For each segment of audio recorded, I forward it to Whisper for transcription. This process results in a semi-live transcription that has been working quite well for me.

However, I'd love to add a bit more control over this process. To that end, I'm looking to create a new variable, MINIMUM_SIGNAL_LENGTH_MSECS. The purpose of this variable is to allow me to dictate the minimum length (like 15 sec) of each recorded segment before it's sent for transcription. 

Here's where I need your help: Could anyone kindly guide me on where in the codebase this new functionality would best be introduced? I'm specifically seeking advice on which function would be the most suitable place to incorporate this feature.

Thank you, looking forward to hearing your ideas.

Best regards,
Abbe
